data-prepper-api/src/test/java/org/opensearch/dataprepper/model/event/JacksonEventTest.java

```diff
@@ -456,6 +456,33 @@ public void testBuild_withStringData() {
         assertThat(event.get("foo", String.class), is(equalTo("bar")));
     }
 
+    @Test
+    public void testBuild_withFormatString() {
+
+        final String jsonString = "{\"foo\": \"bar\", \"info\": {\"ids\": {\"id\":\"idx\"}}}";
+
+        event = JacksonEvent.builder()
+                .withEventType(eventType)
+                .withData(jsonString)
+                .getThis()
+                .build();
+
+        assertThat(event.formatString("test-${foo}-string"), is(equalTo("test-bar-string")));
```

Comment:
Ideally these different tests are not in one test case. You can make all the success tests a parameterized test and it would work quite nicely.

You can use JUnit's `@CsvSource` to separate the actual value from expected. See this documentation: https://junit.org/junit5/docs/5.8.1/api/org.junit.jupiter.params/org/junit/jupiter/params/provider/CsvSource.html#value().

You would not include the negative case with the exception.